remote-wallet/src/trezor.rs

```diff
@@ -0,0 +1,265 @@
+                solana_sign_tx.serialized_tx = Some(data.to_vec());
+                let solana_tx_signature = handle_interaction(
+                    trezor_client
+                        .borrow_mut()
+                        .call(solana_sign_tx, Box::new(|_, m: SolanaTxSignature| Ok(m)))?,
+                )?;
+                Signature::try_from(solana_tx_signature.signature())
+                    .map_err(|_e| RemoteWalletError::Protocol("Signature packet size mismatch"))
+            }
+            _ => Err(RemoteWalletError::NoDeviceFound),
+        }
+    }
+
+    /// Sign off-chain message with wallet managing pubkey at derivation path
+    /// `m/44'/501'/<account>'/<change>'`.
+    fn sign_offchain_message(
+        &self,
+        derivation_path: &DerivationPath,
+        message: &[u8],
+    ) -> Result<Signature, RemoteWalletError> {
+        Self::sign_message(self, derivation_path, message)
+    }
+}
+
+pub fn get_trezor_from_info(
+    info: RemoteWalletInfo,
+    _keypair_name: &str,
+    wallet_manager: &RemoteWalletManager,
+) -> Result<Rc<TrezorWallet>, RemoteWalletError> {
+    wallet_manager.get_trezor_wallet(info.get_pretty_path())
+}
+
+#[cfg(test)]
+mod tests {
+    use {
+        super::*,
+        serial_test::serial,
+        trezor_client::{find_devices, Model},
+    };
+
+    fn init_emulator() -> Trezor {
+        let mut emulator = find_devices(false)
+            .into_iter()
+            .find(|t| t.model == Model::TrezorEmulator)
+            .expect("An emulator should be found")
+            .connect()
+            .expect("Connection to the emulator should succeed");
+        emulator
+            .init_device(None)
+            .expect("Initialization of device should succeed");
+        emulator
+    }
+
+    #[test]
+    #[serial]
+    fn test_emulator_find() {
+        let trezors = find_devices(false);
+        assert!(!trezors.is_empty());
+        assert!(trezors.iter().any(|t| t.model == Model::TrezorEmulator));
+    }
+
+    #[test]
+    #[serial]
+    fn test_solana_pubkey() {
+        let mut emulator = init_emulator();
+        let derivation_path_str = "m/44'/501'/0'/0'";
+        let mut solana_get_pubkey = SolanaGetPublicKey::new();
+        let address_n = convert_path(
+            &bip32::DerivationPath::from_str(derivation_path_str).expect("Failed to parse path"),
+        );
+        solana_get_pubkey.address_n = address_n;
+        solana_get_pubkey.show_display = Some(false);
+        let pubkey = handle_interaction(
+            emulator.call(solana_get_pubkey, Box::new(|_, m: SolanaPublicKey| Ok(m))).expect("Trezor client (the emulator) has been initialized and SolanaGetPublicKey is initialized correctly")
+        ).expect("Trezor client (the emulator) has been initialized and SolanaGetPublicKey is initialized correctly");
+        let public_key_string = bitcoin::base58::encode(pubkey.public_key());
+        assert!(!public_key_string.is_empty());
```

Comment:
How about converting to a `Pubkey` instead? Then we can remove this additional use of tje `bitcoin` crate.